test/optim/test_optim.py

```diff
@@ -834,6 +835,49 @@ def test_rprop(self):
             )
 
 
+    def test_lbfgs(self):
+        self._test_basic_cases(
+            lambda weight, bias: LBFGS([weight, bias]), ignore_multidevice=True
+        )
+        self._test_basic_cases(
+            lambda weight, bias: LBFGS(
+                [weight, bias], line_search_fn="strong_wolfe"
+            ),
+            ignore_multidevice=True,
+        )
+
+
+    def test_lbfgs_complex(self):
```

Comment:
- IMHO it is cleaner to leave it as a separate test than to use conditionals inside test_complex,  the body of the test is different -- they only share some setup lines. So t save a logic duplication of 5 lines, we would have a conditional switching between two different implementations of the inner loop. 
- I tried to match the style of the rest of the file. I could have used` @parametrize`, but this would have been different than in all other test in this file. Not sure what you meant by `@optims` as this would only parameterize for different optimizers, right? 
So I could use `@parametrize / instantiate_parametrized_tests` instead of the loop, or match the current style, or move the existing tests over to parameterize as well.
Not sure what your plans are for the remaining tests, i.e. which ones you  plan to move into the `TestOptimRenewed`

Reply:
`@optims` would allow you to get all the configs for LBFGS so you wouldn't have to specify the forloop toggling linesearch. This way, if new features get added to LBFGS, this test would run automatically for those new configs without changing the test. This would also allow ease of testing across different devices and dtypes. You would want something like

```
@optims([optim for optim in optim_db if optim.optim_cls.__name__ == "LBFGS"], dtypes=[torch.float32])
def test_lbfgs_complex(self, device, dtype)
```

I'm okay with this being a separate test (though I like the idea of mixing in a non-complex param for the other test_complex + would be a change for later). If we want to do that, we should change the skip message in common_optimizers for test_complex for LBFGS to say that it's tested separately through this test.